framework/src/wifi_manager/Kconfig

```diff
@@ -7,11 +7,34 @@ config WIFI_MANAGER
     bool "Enable Wi-Fi Manager"
     default n
 	select NETUTILS_WIFI
-	select WPA_SUPPLICANT
-	select SCSC_WLAN
     ---help---
         Easy APIs for applications to use and control Wi-Fi features
 
+choice
+	prompt "WiFi library"
+	default WPA_SUPPLICANT
+	depends on WIFI_MANAGER
+
+config WPA_SUPPLICANT
```

Comment:
WPA_SUPPLICANT and SCSC_WLAN should be each folder, not here.

How about below?
```
config ARCH_HAVE_WIFI_DRIVER
	bool

config ARCH_HAVE_WPA_SUPPLICANT
	bool

config WIFI_MANAGER
	bool "Enable Wi-Fi Manager"
	default n
	select NETUTILS_WIFI
	select WPA_SUPPLICANT		if !ARCH_HAVE_WPA_SUPPLICANT
	select SCSC_WLAN		if !ARCH_HAVE_WIFI_DRIVER
	select WIFI_3RD_PARTY_FIRMWARE	if ARCH_HAVE_WPA_SUPPLICANT || ARCH_HAVE_WIFI_DRIVER
	---help---
		 Easy APIs for applications to use and control Wi-Fi features

config WIFI_3RD_PARTY_FIRMWARE
	bool

if WIFI_3RD_PARTY_FIRMWARE
# Do it something
endif
```